src/codechicken/nei/recipe/ShapedRecipeHandler.java

```diff
@@ -127,10 +127,15 @@ else if (irecipe instanceof ShapedOreRecipe)
         }
     }
 
+    @SuppressWarnings("unchecked")
     @Override
     public void loadUsageRecipes(ItemStack ingredient) {
         for (IRecipe irecipe : (List<IRecipe>) CraftingManager.getInstance().getRecipeList()) {
             CachedShapedRecipe recipe = null;
+            if (irecipe.getRecipeOutput() == null) {
```

Comment:
I'd move this check down to L146. A lot of `IRecipe` has its recipe output set to null for legitimate reasons, e.g. output is dependent on input and thus doesn't have a static output. This would unnecessarily spam the console with errors.

Reply:
I'm not intending to keep the log in there, I wanted to see what was causing the issue.  forgeShapedRecipe seems to handle exceptions fine, but CachedShapeRecipe will blow up if it has a null output (via PositionedStack --> NEIServerUtils.extractRecipeItems).

Also if a IRecipe has a null output, wouldn't that mean it's not a valid recipe to show up in NEI?

Reply:
all recipes that would be handled by this handlers should not _intentionally_ return null, but here you are checking raw `irecipe`, which could be anything.